core/src/main/java/org/apache/brooklyn/util/core/task/ScheduledTask.java

```diff
@@ -117,6 +121,44 @@ public ScheduledTask(Map<?,?> flags, Callable<Task<?>> taskFactory) {
         cancelOnException = cancelFlag == null || Boolean.TRUE.equals(cancelFlag);
     }
     
+    public static Builder builder(Callable<Task<?>> val) {
+        return new Builder(val);
+    }
+    
+    public static class Builder {
```

Comment:
I've never liked our `ScheduledTask` api. Long term, I'd prefer it to feel more like `java.util.concurrent.ScheduledExecutorService` (rather than it being based on the type of task passed in).

I've not reviewed this properly yet, but I'm guessing the `ScheduledTask.Builder` will make it a bit nicer to work with the existing `ScheduledTask` - so that's a good idea for an incremental improvement. However, I'd eventually like a complete re-write of this area so we can have some kind of executionManager.schedule() method.

Reply:
I can see that. Logically I think the cleanest for both scheduled and one-off tasks would be to separate the submitted job, the execution parameters and metadata (schedule, same thread, name, tags, etc), and the result handle. But not sure how nice that would be to work with. There are times when I want them conflated a la submit(task) then task.getResult but other times I want handle=submit(name, job).

But yes all I did here was make a builder rather than a messy constructor.